pkg/flow/flow.go

```diff
@@ -227,15 +227,18 @@ func (f *Flow) Run(ctx context.Context) {
 			// We need to pop _everything_ from the queue and evaluate each of them.
 			// If we only pop a single element, other components may sit waiting for
 			// evaluation forever.
+
+			// NOTE: sending all the updated nodes at once to EvaluateDependencies
+			//TODO(thampiotr): fix tests and other callers
+			var allUpdated []*controller.ComponentNode
 			for {
 				updated := f.updateQueue.TryDequeue()
```

Comment:
After reading some more code, I think that possibly this approach won't work well, because we'll frequently find only one node in the `updateQueue` and the call to `EvaluateDependencies` is what can result in adding more updated nodes to the queue. For example, relabel components.

I have some other ideas and will be looking into them today.